helm/cluster-azure/templates/_kcp.tpl

```diff
@@ -0,0 +1,138 @@
+{{- define "control-plane" }}
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+kind: KubeadmControlPlane
+metadata:
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+  name: {{ include "resource.default.name" $ }}
+  namespace: {{ $.Release.Namespace }}
+spec:
+  machineTemplate:
+    metadata:
+      labels:
+        {{- include "labels.common" $ | nindent 8 }}
+    infrastructureRef:
+      apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+      kind: AzureMachineTemplate
+      name: {{ include "resource.default.name" $ }}-control-plane
```

Comment:
Some of the fields in cluster api spec are immutable and to be updated requires to actually create a new version of the resource and change the reference. 

Like `AzureMachineTemplate` specs and other fields 

for example in cluster-aws they are adding an hash of the spec to the name of those resources to make sure the update works ( https://github.com/giantswarm/cluster-aws/blob/master/helm/cluster-aws/templates/_control_plane.tpl#L1 ) 

Do you plan to do something similar here ?

Reply:
latest commits already take care of this. will be fixed.